repos/fdbt-site/src/pages/api/multipleOperatorsServiceList.ts

```diff
@@ -1,105 +1,116 @@
-    });
-    return {
-        selectedServices,
-        nocCode,
+    // {SelectedServices} to {MultiOperatorInfo}
+    const multiOperatorsService: SelectedServiceByNocCode[] = [];
+    // export interface SelectedServiceByNocCode {
+    //     [key: string]: ServiceWithNocCode[];
+    // }
+
+    const getIndexOfMultiOperatorsService = (noc: string): number => {
+        const index = multiOperatorsService.findIndex((serviceDetails) => Object.keys(serviceDetails).includes(noc));
+        return index;
     };
+    serviceDetails.forEach((service: ServiceWithNocCode) => {
+        const indexOfFound = service?.nocCode ? getIndexOfMultiOperatorsService(service.nocCode) : -1;
+        if (indexOfFound === -1) {
+            multiOperatorsService.push({ [`${service.nocCode}`]: [service] });
+        } else {
+            if (indexOfFound > -1 && service?.nocCode) {
+                multiOperatorsService[indexOfFound][service.nocCode].push(service);
+            } else {
+                // eslint-disable-next-line no-console
+                console.log(`indexOfFound is ${indexOfFound}`);
+            }
+        }
+    });
+    const newListOfMultiOperatorsData: MultiOperatorInfo[] = [];
+    multiOperatorsService.forEach((obj) => {
+        Object.entries(obj).forEach((v) => {
+            newListOfMultiOperatorsData.push({ nocCode: v[0], services: v[1] });
+        });
+    });
+
+    return newListOfMultiOperatorsData;
 };
 
 export default (req: NextApiRequestWithSession, res: NextApiResponse): void => {
     const redirectUrl = '/multipleOperatorsServiceList';
-    const selectAllText = 'Select All Services';
 
     try {
-        const refererUrl = req?.headers?.referer;
-        const queryString = refererUrl?.substring(refererUrl?.indexOf('?') + 1);
-        const { selectAll } = req.body;
-
-        const isSelected = selectAll === selectAllText;
-
-        if (selectAll && queryString) {
-            redirectTo(res, `${redirectUrl}?selectAll=${isSelected}`);
-            return;
-        }
-        const multiOpDataOnSession = getSessionAttribute(req, MULTIPLE_OPERATORS_SERVICES_ATTRIBUTE);
-        let multiOpDataToReAddToSession: MultiOperatorInfo[] = [];
-        if (isMultiOperatorInfoWithErrors(multiOpDataOnSession)) {
-            multiOpDataToReAddToSession = multiOpDataOnSession.multiOperatorInfo;
-        } else if (multiOpDataOnSession) {
-            multiOpDataToReAddToSession = multiOpDataOnSession;
-        }
-        if ((!req.body || Object.keys(req.body).length === 0) && !selectAll) {
+        if (!req.body || Object.keys(req.body).length === 0) {
             updateSessionAttribute(req, MULTIPLE_OPERATORS_SERVICES_ATTRIBUTE, {
-                multiOperatorInfo: multiOpDataToReAddToSession,
+                multiOperatorInfo: [],
                 errors: [{ id: errorId, errorMessage: 'Choose at least one service from the options' }],
             });
-            redirectTo(res, `${redirectUrl}?selectAll=false`);
```

Comment:
why we are removing this

Reply:
what is removed is the (?selecteAll=false). no we don't use it anymore.
Each service has its own status weather if its selected or not inside the services list.